manifests/service.pp

```diff
@@ -1,19 +1,29 @@
 # configure, enable, and start pulpcore services
 # @api private
 class pulpcore::service {
+
   systemd::unit_file { 'pulpcore-api.service':
     content => template('pulpcore/pulpcore-api.service.erb'),
     active  => true,
     enable  => true,
   }
+
   systemd::unit_file { 'pulpcore-content.service':
     content => template('pulpcore/pulpcore-content.service.erb'),
     active  => true,
     enable  => true,
   }
+
   systemd::unit_file { 'pulpcore-resource-manager.service':
     content => template('pulpcore/pulpcore-resource-manager.service.erb'),
     active  => true,
     enable  => true,
   }
+
+  systemd::unit_file { 'pulpcore-worker@.service':
+    content => template('pulpcore/pulpcore-worker@.service.erb'),
+    active  => true,
```

Comment:
I don't think this works here because the unit is just a template. I think you need to drop the active and enable. Then you can create actual instances with the raw `service` type.